public/js/util/source-map.js

```diff
@@ -0,0 +1,29 @@
+"use strict";
+
+const { networkRequest } = require("./networkRequest");
+const { SourceMapConsumer } = require("source-map");
+
+let sourceMaps = new Map();
+
+function loadSourceMap(tab, source) {
+  const sourceMapURL = `${tab.get("url")}/${source.sourceMapURL}`;
+
+  if (sourceMaps.has(source.id)) {
+    return Promise.resolve(sourceMaps.get(source.id));
+  }
+
+  return networkRequest(sourceMapURL).then(sourceMap => {
+    sourceMaps.set(source.id, sourceMap);
+    return sourceMap;
+  });
+}
+
+function getOriginalSources(sourceMap) {
```

Comment:
We might want to go ahead and nix the generated/original terms, and come up with something better.